### ISSUE TITLE
Get chart element after creation.

### ISSUE THREAD

**Issue by simonsayscode** [krispo/angular-nvd3#232]
Hi is there a way to get the actual element of the chart or parent element of the chart upon successful creation or update? My goal is to get the computed width and height of the chart after it's been created, since I'm leaving the width at the default 100%.

I had thought that this would have something to do with the callback: function (chart) {...} but I'm not sure how to get at the DOM of the chart.

**Comment by krispo** [comment]
Yes, you can access a global directive api using `api` attribute:
```html
<nvd3 options="options" data="data" api="api"></nvd3>
```
and then get chart instance via internal scope:
```js
$scope.api.getScope().chart
```

**Comment by simonsayscode** [comment]
I can access the chart object and the scope, but just curious how to actually get a hold of the DOM object  from that scope so that I can query the width and height of the element.

**Comment by jmsherry** [comment]
`$scope.api.getScope().chart`

That seems to return the chart constructor function?! d3 has a callback, which @simonsayscode is trying to reference - is there any way through to that?

Thanks @krispo ...

**Comment by moonchild100** [comment]
got a related question, from the chart object how can i access the options, like y.range(), y.domain() etc that has been calculated already?
